File: shared/src/main/java/org/tiqr/service/enrollment/ParseEnrollmentChallengeError.java
package org.tiqr.service.enrollment;

import android.os.Bundle;

/**
 * Enrollment error.
 */
public class ParseEnrollmentChallengeError {
    public enum Type {
        UNKNOWN,
        CONNECTION,
        INVALID_CHALLENGE,
        INVALID_RESPONSE
    }

    private final Type _type;
    private final String _title;
    private final String _message;
    private final Bundle _extras;

    /**
     * Constructor.
     *
     * @param type
     * @param message
     */
    public ParseEnrollmentChallengeError(Type type, String title, String message) {
        this(type, title, message, new Bundle());
    }

    /**
     * Constructor.
     *
     * @param type
     * @param message
     * @param extras
     */
    public ParseEnrollmentChallengeError(Type type, String title, String message, Bundle extras) {
        _type = type;
        _title = title;
        _message = message;
        _extras = extras;
    }

    /**
     * Error type.
     *
     * @return Error type.
     */
    public Type getType() {
        return _type;
    }

    /**
     * Title.
     *
     * @return Error title.
     */
    public String getTitle() {
        return _title;
    }

    /**
     * Message.
     *
     * @return Error message.
     */
    public String getMessage() {
        return _message;
    }

    /**
     * Extras.
     *
     * @return Extras.
     */
    public Bundle getExtras() {
        return _extras;
    }
}

File: shared/src/main/java/org/tiqr/service/enrollment/ParseEnrollmentChallengeError.kt
package org.tiqr.service.enrollment

import android.os.Bundle

/**
 * Enrollment error.
 */
class ParseEnrollmentChallengeError
/**
 * Constructor.
 *
 * @param type
 * @param message
 * @param extras
 */
@JvmOverloads constructor(
        /**
         * Error type.
         *
         * @return Error type.
         */
        val type: Type,
        /**
         * Title.
         *
         * @return Error title.
         */
        val title: String,
        /**
         * Message.
         *
         * @return Error message.
         */
        val message: String,
        /**
         * Extras.
         *
         * @return Extras.
         */
        val extras: Bundle = Bundle()) {
    enum class Type {
        UNKNOWN,
        CONNECTION,
        INVALID_CHALLENGE,
        INVALID_RESPONSE
    }
}
/**
 * Constructor.
 *
 * @param type
 * @param message
 */
